core/domain/values/assets/StylesheetAsset.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace EventEspresso\core\domain\values\assets;
+
+use EventEspresso\core\domain\DomainInterface;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+
+/**
+ * Class StylesheetAsset
+ * Details for a Cascading Stylesheet asset
+ *
+ * @package EventEspresso\core\domain\values\assets
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class StylesheetAsset extends BrowserAsset
+{
+
+    /**
+     * @var string $media
+     */
+    private $media;
+
+
+    /**
+     * CssFile constructor.
+     *
+     * @param                 $handle
+     * @param string          $source
+     * @param array           $dependencies
+     * @param                 $media
+     * @param DomainInterface $domain
+     * @throws InvalidDataTypeException
+     */
+    public function __construct($handle, $source, array $dependencies, $media, DomainInterface $domain)
```

Comment:
should there be a default for the $media argument?

Reply:
a default is declared in `\EventEspresso\core\services\assets\AssetManager::addStylesheet()` which would be the preferred way to declare assets since that method also takes care of adding them to the `AssetCollection` class.
I hadn't previously realized that the PSR-2 guidelines state:

> Method arguments with default values MUST go at the end of the argument list.

which throws a bit of a wrench into how we are using our DI container and auto-injected dependencies, which I have been suggesting to always place as the last constructor arguments because otherwise you would have to manually supply them when overriding a default argument.

For example, if we wanted the `StylesheetAsset` class to declare a default for `$media` **AND** not violate PSR-2, then its signature would need to be:

```php
public function __construct($handle, $source, array $dependencies, DomainInterface $domain, $media='all')
```

which would then mean that any client code would then be responsible for providing an instance of `DomainInterface`. 

The following code:

```php
$default = $this->loader->getNew(
    'EventEspresso\core\domain\values\assets\StylesheetAsset',
    array(
        'default-css',
        EE_GLOBAL_ASSETS_URL . 'css/espresso_default.css',
        array('dashicons'),
        null,
        'screen'
    )
);
```

would result in the following:

> Fatal error: Uncaught TypeError: Argument 4 passed to EventEspresso\core\domain\values\assets\StylesheetAsset::__construct() must implement interface EventEspresso\core\domain\DomainInterface, null given

The reason for this is because the DI container sees that you have supplied a fourth argument in the array of values passed to the Loader, despite that value being `null`. It can not differentiate between a value of `null` representing the desire to use the default value for an argument, and an actual value of `null`.
Of course, this can probably be easily overcome by introducing a constant to indicate that the default value for an argument should be used instead of null, 
ie: something like:

```php
$default = $this->loader->getNew(
    'EventEspresso\core\domain\values\assets\StylesheetAsset',
    array(
        'default-css',
        EE_GLOBAL_ASSETS_URL . 'css/espresso_default.css',
        array('dashicons'),
        Loader::USE_DEFAULT_VALUE,
        'screen'
    )
);
```

which is actually a really good idea, and something I should probably implement regardless.

As well, if we are going to add a default value for `$media` for this class, then we should likely also add an empty array as the default value for `$dependencies`